domain/personallearningplan/src/test/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/service/GoalServiceTest.kt

```diff
@@ -150,83 +151,296 @@ class GoalServiceTest {
-    assertThat(exception.goalReference).isEqualTo(goalReference)
-    verify(goalPersistenceAdapter).getGoal(prisonNumber, goalReference)
+      // Then
+      assertThat(exception).hasMessage("Goal with reference [$goalReference] for prisoner [$prisonNumber] not found")
+      assertThat(exception.prisonNumber).isEqualTo(prisonNumber)
+      assertThat(exception.goalReference).isEqualTo(goalReference)
+      verify(goalPersistenceAdapter).getGoal(prisonNumber, goalReference)
+    }
   }
 
-  @Test
-  fun `should update goal given goal exists`() {
-    // Given
-    val prisonNumber = randomValidPrisonNumber()
-    val goalReference = aValidReference()
+  @Nested
+  inner class UpdateGoal {
+    @Test
+    fun `should update goal given goal exists`() {
+      // Given
+      val prisonNumber = randomValidPrisonNumber()
+      val goalReference = aValidReference()
 
-    val goal = aValidGoal(reference = goalReference)
-    given(goalPersistenceAdapter.getGoal(any(), any())).willReturn(goal)
-    given(goalPersistenceAdapter.updateGoal(any(), any())).willReturn(goal)
+      val goal = aValidGoal(reference = goalReference)
+      given(goalPersistenceAdapter.getGoal(any(), any())).willReturn(goal)
+      given(goalPersistenceAdapter.updateGoal(any(), any())).willReturn(goal)
 
-    val updatedGoal = aValidUpdateGoalDto(reference = goalReference)
+      val updatedGoal = aValidUpdateGoalDto(reference = goalReference)
 
-    // When
-    val actual = service.updateGoal(prisonNumber, updatedGoal)
+      // When
+      val actual = service.updateGoal(prisonNumber, updatedGoal)
 
-    // Then
-    assertThat(actual).isEqualTo(goal)
-    verify(goalPersistenceAdapter).updateGoal(prisonNumber, updatedGoal)
-    verify(goalEventService).goalUpdated(prisonNumber, goal, actual)
-  }
+      // Then
+      assertThat(actual).isEqualTo(goal)
+      verify(goalPersistenceAdapter).updateGoal(prisonNumber, updatedGoal)
+      verify(goalEventService).goalUpdated(prisonNumber, goal, actual)
+    }
+
+    @Test
+    fun `should not update goal given goal does not exist`() {
+      // Given
+      val prisonNumber = randomValidPrisonNumber()
+      val goalReference = aValidReference()
+
+      given(goalPersistenceAdapter.getGoal(any(), any())).willReturn(null)
+
+      val updatedGoal = aValidUpdateGoalDto(reference = goalReference)
+
+      // When
+      val exception =
+        catchThrowableOfType(GoalNotFoundException::class.java) { service.updateGoal(prisonNumber, updatedGoal) }
+
+      // Then
+      assertThat(exception).hasMessage("Goal with reference [$goalReference] for prisoner [$prisonNumber] not found")
+      assertThat(exception.prisonNumber).isEqualTo(prisonNumber)
+      assertThat(exception.goalReference).isEqualTo(goalReference)
+      verify(goalPersistenceAdapter, never()).updateGoal(prisonNumber, updatedGoal)
+      verifyNoInteractions(goalNotesService)
+      verifyNoInteractions(goalEventService)
+    }
+
+    @Nested
```

Comment:
These are the new tests for updating notes as part of updating the goal

(the rest of the change in this test class is just a little refactoring to organise some of the existing tests into inner classes 👍 )